### PR TITLE
Render stats

### LINKED ISSUE #17: Render Stats
Display an ImGui window with some basic rendering stats

- [x] Frametime/FPS
- [x] Tris being rendered
- [x] Draw calls

### PR DESCRIPTION
**Pull request by hillj35**
Tracks a few render stats (frametime/fps, tri count, draw calls)

also updates the view space multiplication for the directional light direction vector

Screenshot from 2024-05-05 11-48-46

closes #17

### CODE REVIEW

File: src/ic_renderer.cpp [hunk 1]
```diff
@@ -5,7 +5,9 @@
 #include <algorithm>
 
 namespace IC {
-    Renderer::Renderer(const RendererConfig &config) : window(config.window) {}
+    Renderer::Renderer(const RendererConfig &config) : window(config.window) {
+        AddImguiFunction(std::bind(&Renderer::RenderStatsGUI, this));
```
Comment (DevWizaad) [inline src/ic_renderer.cpp]: Does this need to be removed in the destructor?

File: src/vulkan/vulkan_renderer.cpp [hunk 2]
```diff
@@ -73,6 +73,8 @@ namespace IC {
     }
 
     void VulkanRenderer::DrawFrame() {
+        auto start = std::chrono::system_clock::now();
```
Comment (DevWizaad) [inline src/vulkan/vulkan_renderer.cpp]: `std::chrono::high_resolution_clock` would be more accurate here. I think on Windows `system_clock` only has millisecond precision.

Comment (Colbydude) [reply src/vulkan/vulkan_renderer.cpp]: GLFW also has `glfwGetTime()`, that might be better for a cross-platform approach. Not sure on the precision on it.

File: src/ic_renderer.h [hunk 3]
```diff
@@ -32,15 +38,19 @@ namespace IC {
...
+        std::unordered_map<std::string, std::function<void()>> imGuiFunctions;
         GLFWwindow *window;
+        RenderStats renderStats{};
+
+        void RenderStatsGUI();
 
     private:
         Renderer(const Renderer &) = delete;
         Renderer &operator=(const Renderer &) = delete;
+        const std::string STATS_WINDOW_NAME = "render stats";
```
Comment (DevWizaad) [inline src/ic_renderer.h:54]: Can you make this a `static constexpr`? I don't remember if that's allowed for `std::string`s but if not can you at least make it `static const`?

**Review by DevWizaad — APPROVED** [review]

**Pull request merged**